src/Symfony/Component/Mailer/Transport/AbstractTransport.php

```diff
@@ -103,6 +111,13 @@ public function send(RawMessage $message, ?Envelope $envelope = null): ?SentMess
         }
     }
 
+    public function setRateLimiterFactory(RateLimiterFactoryInterface $rateLimiterFactory): static
+    {
+        $this->rateLimiterFactory = $rateLimiterFactory;
+
+        return $this;
```

Comment:
I don't think we should return `$this` here. It makes the API less clear whether this is a method mutating the instance or returning a new one. Our setters generally don't return `$this` in Symfony.

Reply:
The setters in `Symfony\Component\Mailer\Transport\AbstractTransport` all return `$this` - so I think we should also return `$this` here.

Reply:
Keeping as is, as `AbstractTransport` always returns `$this`.